### PR TITLE
[Ep-246] Page Viewed discover

### PR DESCRIPTION
**Pull request by tobitech**
# 📲 What

Modify tracking method that sends event when user loads any of the explore tabs

# 🤔 Why

Convert Data Lake event to segment

### CODE REVIEW

**Review by singhhari — CHANGES_REQUESTED** [review]
Works as expected so no issues there but we need to bump the coverage.

File: Library/ViewModels/DiscoveryPageViewModel.swift [hunk 1]
```diff
@@ -415,14 +415,9 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
...
 
-    requestFirstPageWith
-      .observeValues { params in
-        let optimizelyProps = optimizelyProperties() ?? [:]
-
-        AppEnvironment.current.ksrAnalytics.trackDiscovery(
-          params: params,
-          optimizelyProperties: optimizelyProps
-        )
+    Signal.combineLatest(requestFirstPageWith, self.viewWillAppearProperty.signal)
```
Comment (singhhari) [inline Library/ViewModels/DiscoveryPageViewModel.swift:418]: How do you feel about something like this:
`    requestFirstPageWith.signal
      .takeWhen(self.viewWillAppearProperty.signal)
      .observeValues { params in
        AppEnvironment.current.ksrAnalytics.trackDiscovery(params: params)
      }`

Removes the _ in the observeValues block. Test it and see if it gives the same desired effect.

File: Library/Tracking/KSRAnalytics.swift [hunk 2]
```diff
@@ -1648,7 +1645,7 @@ private func discoveryProperties(
     .withAllValuesFrom(parentCategoryProps ?? [:])
 
   result["everything"] = result.isEmpty
-  result["sort"] = params.sort?.rawValue
+  result["sort"] = params.sort?.trackingString
```
Comment (singhhari) [inline Library/Tracking/KSRAnalytics.swift:1648]: Hm why did this change?

**Review by singhhari — CHANGES_REQUESTED** [review]

File: Library/Tracking/KSRAnalytics.swift (hunk 2 above)
Comment (tobitech) [reply Library/Tracking/KSRAnalytics.swift:1648]: I realized the sort values that we were sending to segment was using the rawValue which wasn't correct. The ticket requires the string values held by the trackingString property. It was until I tested for other enum cases that I found out

Comment (singhhari) [reply Library/Tracking/KSRAnalytics.swift:1648]: good catch

**Review by singhhari — APPROVED** [review]
lgtm

**Pull request merged**